servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -463,13 +472,18 @@ public long createFile(TachyonURI path, long blockSizeBytes, boolean recursive)
       List<Inode> created =
           createFileInternal(path, blockSizeBytes, recursive, System.currentTimeMillis());
 
-      // Journal all of the created inodes.
-      for (Inode inode : created) {
-        writeJournalEntry(inode.toJournalEntry());
-      }
       writeJournalEntry(mDirectoryIdGenerator.toJournalEntry());
+      for (int i = 0; i < created.size(); i ++) {
+        Inode inode = created.get(i);
+        if (i == 0) {
+          // The first inode in the list was modified, not created.
```

Comment:
Yeah... I am conflicted about this. Should I just rename this as "modified" with the implication that the first one is modified, and the rest are created, or do I create a separate return data structure to encapsulate the modified one, and the created ones.

I guess I could also use a Pair<>, but for some reason, I don't really like Pair.

What do you suggest?